content/patterns/button/button-pattern.html

```diff
@@ -14,93 +14,94 @@
-                For example, activating a cancel button in a dialog returns focus to the button that opened the dialog.
-                However, if the dialog were confirming the action of deleting the page from which it was opened, the focus would logically move to a new context.
+                ボタンを動作させるとダイアログが閉じる場合、ダイアログのコンテキストで実行された機能が論理的に異なる要素につながる場合を除き、フォーカスは通常、ダイアログを開いたボタンに戻ります。
+                たとえば、ダイアログでキャンセルボタンを動作させると、フォーカスはダイアログを開いたボタンに戻ります。
+                ただし、ダイアログが開かれたページを削除するアクションを確認している場合、フォーカスは論理的に新しいコンテキストに移動します。
               </li>
-              <li>If activating the button does not dismiss the current context, then focus typically remains on the button after activation, e.g., an Apply or Recalculate button.</li>
-              <li>If the button action indicates a context change, such as move to next step in a wizard or add another search criteria, then it is often appropriate to move focus to the starting point for that action.</li>
+              <li>ボタンを動作させても現在のコンテキストが閉じない場合、通常、動作後もフォーカスはボタンに残ります。例：適用ボタン又は再計算ボタン。</li>
+              <li>ボタンのアクションが、ウィザードの次のステップへの移動や別の検索条件の追加など、コンテキストの変化を示す場合、そのアクションの開始点にフォーカスを移動するのが適切な場合が多いです。</li>
               <li>
-                If the button is activated with a shortcut key, the focus usually remains in the context from which the shortcut key was activated.
-                For example, if <kbd>Alt</kbd> + <kbd>U</kbd> were assigned to an &quot;Up&quot; button that moves the currently focused item in a list one position higher in the list, pressing <kbd>Alt</kbd> + <kbd>U</kbd> when the focus is in the list would not move the focus from the list.
+                ショートカットキーでボタンを動作させた場合、フォーカスは通常、ショートカットキーを動作させたコンテキストに残ります。
+                たとえば、リスト内の現在フォーカスされている項目をリスト内で 1 つ上の位置に移動する「上へ」ボタンに <kbd>Alt</kbd> + <kbd>U</kbd> が割り当てられている場合、リストにフォーカスがあるときに <kbd>Alt</kbd> + <kbd>U</kbd> を押しても、リストからフォーカスは移動しません。
               </li>
             </ul>
           </li>
         </ul>
       </section>
 
       <section id="roles_states_properties">
-        <h2>WAI-ARIA Roles, States, and Properties</h2>
+        <h2>WAI-ARIA ロール、ステート、及びプロパティ</h2>
         <ul>
-          <li>The button has role of <a class="role-reference" href="#button">button</a>.</li>
+          <li>ボタンは <a class="role-reference" href="#button">button</a> のロールを持ちます。</li>
           <li>
-            The <code>button</code> has an accessible label.
-            By default, the accessible name is computed from any text content inside the button element.
-            However, it can also be provided with <a href="#aria-labelledby" class="property-reference">aria-labelledby</a> or <a href="#aria-label" class="property-reference">aria-label</a>.
+            <code>button</code> はアクセシブルなラベルを持ちます。
+            デフォルトでは、アクセシブルネームはボタン要素内のテキストコンテンツから算出されます。
```

Comment:
accessible name の訳ですが、WCAG 関連文書の翻訳では「アクセシブルな名前」と訳しているので、それと合わせるのがよいのかなと思いました。以下のような感じで、いかがでしょうか。

```suggestion
            デフォルトでは、アクセシブルな名前は、ボタン要素内のテキストコンテンツから算出されます。
```